### PR TITLE
mutate: Deduplicering

### PR DESCRIPTION
**Pull request by virrenn**
Changed so the report uses mainly the status ID instead of the mutation ID, it will decrease the amount of duplicate mutants

Viktor

### CODE REVIEW

File: plugin/mutate/source/dextool/plugin/mutate/backend/database/package.d [hunk 1]
```diff
@@ -183,6 +184,7 @@ struct Database {
     void iterateMutants(const Mutation.Kind[] kinds, void delegate(const ref IterateMutantRow2) dg) @trusted {
         immutable sql = format("SELECT
             t0.id,
```
Comment (joakim-brannstrom-work1) [inline plugin/mutate/source/dextool/plugin/mutate/backend/database/package.d:186]: t0.id is unused? if so remove it

File: plugin/mutate/source/dextool/plugin/mutate/backend/database/type.d [hunk 2]
```diff
@@ -60,6 +60,7 @@ struct MutationEntry {
     MutationPoint mp;
     MutantTimeProfile profile;
     Language lang;
+    MutationStatusId st_id;
```
Comment (joakim-brannstrom-work1) [inline plugin/mutate/source/dextool/plugin/mutate/backend/database/type.d:63]: change `st_id` to `stId` to follow the name standard.
sql queries are snake_case while D code is camelCase.

File: plugin/mutate/source/dextool/plugin/mutate/backend/database/type.d [hunk 3]
```diff
@@ -121,6 +122,7 @@ struct MutantInfo {
     ExitStatus ecode;
     Mutation.Kind kind;
     SourceLoc sloc;
+    MutationStatusId st_id;
```
Comment (joakim-brannstrom-work1) [inline plugin/mutate/source/dextool/plugin/mutate/backend/database/type.d:125]: stId

File: plugin/mutate/source/dextool/plugin/mutate/backend/report/html/package.d [hunk 4]
```diff
@@ -191,15 +191,17 @@ nothrow:
     }
 
     MutationId id;
+    MutationStatusId st_id;
```
Comment (joakim-brannstrom-work1) [inline plugin/mutate/source/dextool/plugin/mutate/backend/report/html/package.d:194]: stId

File: plugin/mutate/source/dextool/plugin/mutate/backend/report/html/package.d [hunk 5]
```diff
@@ -191,15 +191,17 @@ nothrow:
...
     Offset offset;
     Text txt;
     Mutation mut;
 
-    this(MutationId id, Offset offset, string original, string mutation, Mutation mut) {
+    this(MutationId id, MutationStatusId st_id, Offset offset, string original, string mutation, Mutation mut) {
```
Comment (joakim-brannstrom-work1) [inline plugin/mutate/source/dextool/plugin/mutate/backend/report/html/package.d:199]: stId

**Comment by joakim-brannstrom-work1** [comment]
run autoformat on the source code :)

File: plugin/mutate/source/dextool/plugin/mutate/backend/database/type.d (hunk 2 above)
Comment (virrenn) [reply plugin/mutate/source/dextool/plugin/mutate/backend/database/type.d:63]: Should I change all the places where you use snake_case too in the D code or just where I use it?

**Pull request merged**